nemo_aligner/utils/train_script_utils.py

```diff
@@ -223,3 +223,56 @@ def temp_pop_from_config(cfg, name_to_temporarily_remove):
             yield
         finally:
             setattr(cfg, name_to_temporarily_remove, temp)
+
+
+def compute_mbs(num_rollout_samples, rollout_micro_batch_size, num_rollout_per_prompt, data_parallel_world_size):
+    '''
+    Batch Size Logic:
+
+    User Specifies:
+    - rollout_micro_batch_size: Number of generations we are able to fit on one worker
+    - num_rollout_samples: This is the number of generations we want to generate in each pass
+    - num_rollout_per_prompt: The number of rollout samples for any given prompt
+
+    We need to get:
+    - mbs: The number of prompts to load per worker
+    - duplicate: The number of times to duplicate each prompt
+    - generation_iter: number of iterations to run generation for
+    
+    if num_rollout_per_prompt * 2 < rollout_micro_batch_size:
```

Comment:
« O: » It is unclear where this two factor is coming from

Reply:
mbs=1 can lead to the error "The provided qkv memory layout is not supported!". So for now, I ensure mbs >= 2.

Reply:
> "The provided qkv memory layout is not supported!"

This unfortunately is hardware dependent -> I don't think you get this error on H100/A100s but probably your workstation. I tend to think this is not our issue to fix so I'm ok with not ensuring mbs >= 2 for simplicity.

Reply:
it seems like once you relax this constraint this function is not needed?